sopel/irc/isupport.py

```diff
@@ -0,0 +1,380 @@
+# coding=utf-8
+"""IRC Tools for ISUPPORT management.
+
+When a server wants to advertise its features and settings, it can use the
+``RPL_ISUPPORT`` command with a list of arguments.
+
+.. seealso::
+
+    https://modern.ircdocs.horse/#rplisupport-005
+
+"""
+# Copyright 2019, Florian Strzelecki <florian.strzelecki@gmail.com>
+#
+# Licensed under the Eiffel Forum License 2.
+from __future__ import unicode_literals, absolute_import, print_function, division
+
+import functools
+import itertools
+import re
+
+
+def _optional(parser, default=None):
```

Comment:
For this and all of the "private" parser methods, it would be more consistent with the overhauled `irc` submodule docs (see #1772) to have proper docstrings as if Sphinx will generate docs based on them, even though they aren't presently included in any public pages.